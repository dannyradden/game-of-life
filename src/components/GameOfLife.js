import React, { Component } from 'react';
import Tile from './Tile';

class GameOfLife extends Component {
  constructor(props) {
    super(props);
    this.state = {};

    this.buttonElements = {};
  }

  renderRow() {
    let items = [];
    for (var i = 1; i < this.props.length + 1; i++) {
      items.push(
        <Tile
          key={i}
          name={'tile' + i}
          buttonRef={el => {
            if (!el) return;
            this.buttonElements[el.name] = el;
          }}
          value="0"
        />
      );
    }

    return items;
  }

  render() {
    return <div className="container">{this.renderRow()}</div>;
  }
}

export default GameOfLife;
