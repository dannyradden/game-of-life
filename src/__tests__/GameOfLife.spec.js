import React from 'react';
import { shallow } from 'enzyme';
import GameOfLife from '../components/GameOfLife';

test('GameOfLife renders correctly', () => {
  const component = shallow(<GameOfLife rows={2} columns={2} seed={[0, 1, 0, 1]} />);
  expect(component).toMatchSnapshot();
});

test('genArray uses seed if correct length', () => {
  const seed = [0, 1, 0, 1];
  const component = shallow(<GameOfLife rows={2} columns={2} seed={seed} />);
  expect(component.state().genArray).toEqual(seed);
});

test('genArray doesnt use seed if correct length', () => {
  const seed = [0, 1, 0];
  const component = shallow(<GameOfLife rows={2} columns={2} seed={seed} />);
  expect(component.state().genArray).not.toEqual(seed);
});
