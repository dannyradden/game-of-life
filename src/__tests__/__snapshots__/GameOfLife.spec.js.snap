// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOfLife renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameOfLife
    columns={2}
    rows={2}
    seed={
        Array [
            0,
            1,
            0,
            1,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="grid"
>
          <div>
                    <Tile
                              buttonRef={[Function]}
                              dataId={0}
                              handleClick={[Function]}
                              name="tile1"
                              value={0}
                    />
                    <Tile
                              buttonRef={[Function]}
                              dataId={1}
                              handleClick={[Function]}
                              name="tile2"
                              value={1}
                    />
          </div>
          <div>
                    <Tile
                              buttonRef={[Function]}
                              dataId={2}
                              handleClick={[Function]}
                              name="tile3"
                              value={0}
                    />
                    <Tile
                              buttonRef={[Function]}
                              dataId={3}
                              handleClick={[Function]}
                              name="tile4"
                              value={1}
                    />
          </div>
</div>,
        <button
          onClick={[Function]}
>
          Next Generation!
</button>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <Tile
                            buttonRef={[Function]}
                            dataId={0}
                            handleClick={[Function]}
                            name="tile1"
                            value={0}
              />
              <Tile
                            buttonRef={[Function]}
                            dataId={1}
                            handleClick={[Function]}
                            name="tile2"
                            value={1}
              />
</div>,
            <div>
              <Tile
                            buttonRef={[Function]}
                            dataId={2}
                            handleClick={[Function]}
                            name="tile3"
                            value={0}
              />
              <Tile
                            buttonRef={[Function]}
                            dataId={3}
                            handleClick={[Function]}
                            name="tile4"
                            value={1}
              />
</div>,
          ],
          "className": "grid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Tile
                  buttonRef={[Function]}
                  dataId={0}
                  handleClick={[Function]}
                  name="tile1"
                  value={0}
/>,
                <Tile
                  buttonRef={[Function]}
                  dataId={1}
                  handleClick={[Function]}
                  name="tile2"
                  value={1}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "buttonRef": [Function],
                  "dataId": 0,
                  "handleClick": [Function],
                  "name": "tile1",
                  "value": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "buttonRef": [Function],
                  "dataId": 1,
                  "handleClick": [Function],
                  "name": "tile2",
                  "value": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Tile
                  buttonRef={[Function]}
                  dataId={2}
                  handleClick={[Function]}
                  name="tile3"
                  value={0}
/>,
                <Tile
                  buttonRef={[Function]}
                  dataId={3}
                  handleClick={[Function]}
                  name="tile4"
                  value={1}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "buttonRef": [Function],
                  "dataId": 2,
                  "handleClick": [Function],
                  "name": "tile3",
                  "value": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "class",
                "props": Object {
                  "buttonRef": [Function],
                  "dataId": 3,
                  "handleClick": [Function],
                  "name": "tile4",
                  "value": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next Generation!",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next Generation!",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="grid"
>
            <div>
                        <Tile
                                    buttonRef={[Function]}
                                    dataId={0}
                                    handleClick={[Function]}
                                    name="tile1"
                                    value={0}
                        />
                        <Tile
                                    buttonRef={[Function]}
                                    dataId={1}
                                    handleClick={[Function]}
                                    name="tile2"
                                    value={1}
                        />
            </div>
            <div>
                        <Tile
                                    buttonRef={[Function]}
                                    dataId={2}
                                    handleClick={[Function]}
                                    name="tile3"
                                    value={0}
                        />
                        <Tile
                                    buttonRef={[Function]}
                                    dataId={3}
                                    handleClick={[Function]}
                                    name="tile4"
                                    value={1}
                        />
            </div>
</div>,
          <button
            onClick={[Function]}
>
            Next Generation!
</button>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Tile
                                buttonRef={[Function]}
                                dataId={0}
                                handleClick={[Function]}
                                name="tile1"
                                value={0}
                />
                <Tile
                                buttonRef={[Function]}
                                dataId={1}
                                handleClick={[Function]}
                                name="tile2"
                                value={1}
                />
</div>,
              <div>
                <Tile
                                buttonRef={[Function]}
                                dataId={2}
                                handleClick={[Function]}
                                name="tile3"
                                value={0}
                />
                <Tile
                                buttonRef={[Function]}
                                dataId={3}
                                handleClick={[Function]}
                                name="tile4"
                                value={1}
                />
</div>,
            ],
            "className": "grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Tile
                    buttonRef={[Function]}
                    dataId={0}
                    handleClick={[Function]}
                    name="tile1"
                    value={0}
/>,
                  <Tile
                    buttonRef={[Function]}
                    dataId={1}
                    handleClick={[Function]}
                    name="tile2"
                    value={1}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "buttonRef": [Function],
                    "dataId": 0,
                    "handleClick": [Function],
                    "name": "tile1",
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "buttonRef": [Function],
                    "dataId": 1,
                    "handleClick": [Function],
                    "name": "tile2",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Tile
                    buttonRef={[Function]}
                    dataId={2}
                    handleClick={[Function]}
                    name="tile3"
                    value={0}
/>,
                  <Tile
                    buttonRef={[Function]}
                    dataId={3}
                    handleClick={[Function]}
                    name="tile4"
                    value={1}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "buttonRef": [Function],
                    "dataId": 2,
                    "handleClick": [Function],
                    "name": "tile3",
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "class",
                  "props": Object {
                    "buttonRef": [Function],
                    "dataId": 3,
                    "handleClick": [Function],
                    "name": "tile4",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next Generation!",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next Generation!",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
